bofire/strategies/doe/design.py

```diff
@@ -583,53 +605,76 @@ def partially_fix_experiment(
 
 
 def check_fixed_experiments(
-    domain: Domain, n_experiments: int, fixed_experiments: np.ndarray
+    domain: Domain, n_experiments: int, fixed_experiments: pd.DataFrame
 ) -> None:
     """Checks if the shape of the fixed experiments is correct and if the number of fixed experiments is valid
     Args:
         domain (Domain): domain defining the input variables used for the check.
         n_experiments (int): total number of experiments in the design that fixed_experiments are part of.
-        fixed_experiments (np.ndarray): fixed experiment proposals to be checked.
+        fixed_experiments (pd.DataFrame): fixed experiment proposals to be checked.
     """
 
-    n_fixed_experiments, D = np.array(fixed_experiments).shape
+    n_fixed_experiments = len(fixed_experiments.index)
 
     if n_fixed_experiments >= n_experiments:
         raise ValueError(
             "For starting the optimization the total number of experiments must be larger that the number of fixed experiments."
         )
 
-    if D != len(domain.inputs):
+    input_set = set(domain.inputs.get_keys())
```

Comment:
You could also somehow solve this by `domain.validate_experiments(only_inputs=True)`

Reply:
Did this. Required to change the validate_candidates function slightly (add additional flags) as `partially_fixed_experiments` will contain `None`.

Reply:
Reverted these changes. Only use `validate_candidates` for `fixed_experiments` now - where the function can stay as it is.